packages/calcite-components/src/components/list/list.tsx

```diff
@@ -800,9 +806,9 @@ export class List implements InteractiveComponent, LoadableComponent, SortableCo
     this.connectObserver();
 
     this.calciteListOrderChange.emit({
-      dragEl: sortItem,
-      fromEl: parentEl,
-      toEl: parentEl,
+      dragEl: sortItem as HTMLCalciteListItemElement,
```

Comment:
I think you can avoid type casting here if you type `parentEl` when it's assigned.